sdk/storage/azure-storage-quickquery/src/main/java/com/azure/storage/quickquery/BlobQuickQueryInputStream.java

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.quickquery;
+
+import com.azure.storage.common.ErrorReceiver;
+import com.azure.storage.common.ProgressReceiver;
+import com.azure.storage.quickquery.models.BlobQuickQueryError;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+/**
+ * An implementation of {@link InputStream} that allows users to read blob quick query responses.
+ */
+public class BlobQuickQueryInputStream extends InputStream {
+
+    // The stream that returns parsed avro
+    private DataFileStream<GenericRecord> parsedStream;
+
+    // The internal user buffer used to hold parsed data.
+    private ByteArrayInputStream buffer;
+
+    // Whether or not the end record has been seen or not.
+    private boolean endRecordSeen;
+
+    // The raw avro stream coming from the network.
+    private InputStream networkStream;
+
+    // User provided error and progress receivers.
+    private final ErrorReceiver<BlobQuickQueryError> errorReceiver;
+    private final ProgressReceiver progressReceiver;
+
+    /**
+     * Creates a new {@code BlobQuickQueryInputStream}
+     * @param networkStream {@link InputStream network stream}
+     * @param errorReceiver {@link ErrorReceiver<BlobQuickQueryError>}
+     * @param progressReceiver {@link ProgressReceiver}
+     */
+    BlobQuickQueryInputStream(InputStream networkStream, ErrorReceiver<BlobQuickQueryError> errorReceiver,
```

Comment:
nit: naming. This class doesn't seem to assume that stream it's wrapping is coming from network. That's piece of knowledge caller of this constructor would have.